pkg/controller/instance/instance_controller.go

```diff
@@ -376,24 +423,46 @@ type ReconcileInstance struct {
 // +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=kudo.dev,resources=instances,verbs=get;list;watch;create;update;patch;delete
 func (r *ReconcileInstance) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	ctx := context.TODO()
 	// Fetch the Instance instance
-	instance := &kudov1alpha1.Instance{}
-	err := r.Get(context.TODO(), request.NamespacedName, instance)
-	if err != nil {
-		if errors.IsNotFound(err) {
-			// Object not found, return.  Created objects are automatically garbage collected.
-			// For additional cleanup logic use finalizers.
-			return reconcile.Result{}, nil
-		}
-		// Error reading the object - requeue the request.
+	// Fetch the Instance instance
+	instance, err := getInstance(ctx, r, request)
+	if err != nil || instance == nil {
```

Comment:
I think instance cannot be nil here if err is empty...

Reply:
It seemed cleaner to me to have a function do all the work of getting an instance... the underlying details are likely understood.... however I've maintained the original logic...   this is a search for an instance...  err != nil if there is an error... instance != nil if one is found... however if no instance is found that is not an error so the check for instance == nil is necessary.